### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by jacky309**

### CODE REVIEW

File: cmake/faceliftMacros.cmake [hunk 1]
```diff
@@ -299,27 +300,32 @@ function(facelift_add_interface TARGET_NAME)
 
     endif()
 
+    if(ARGUMENT_MONOLITHIC_SUPPORTED)
+        set(ARGUMENT_MONOLITHIC MONOLITHIC_SUPPORTED)
+        message("OOOO ${LIBRARY_NAME}")
```
Comment (bitmouse) [inline cmake/faceliftMacros.cmake]: Are you sure you want that kind of message?

File: cmake/faceliftMacros.cmake [hunk 2]
```diff
@@ -299,27 +300,32 @@ function(facelift_add_interface TARGET_NAME)
...
-                $<BUILD_INTERFACE:${IPC_DBUS_OUTPUT_PATH}>
-        )
-    endif()
-
-    target_compile_definitions(${LIBRARY_NAME} PRIVATE "${MODULE_COMPILE_DEFINITIONS}")
-
     # Get the list of files from the interface definition folder so that we can regenerate the code whenever there is a change there
     file(GLOB_RECURSE QFACE_FILES ${ARGUMENT_INTERFACE_DEFINITION_FOLDER}/*.qface)
-    target_sources(${LIBRARY_NAME} PRIVATE ${QFACE_FILES})
+#    target_sources(${LIBRARY_NAME} PRIVATE INTERFACE ${QFACE_FILES})
```
Comment (bitmouse) [inline cmake/faceliftMacros.cmake]: Please don't leave commented out lines. Remove it or add reason why it is commented out.

File: cmake/faceliftMacros.cmake [hunk 3]
```diff
@@ -376,11 +383,15 @@ macro(_facelift_add_target_start additionalOptions additionalOneValueArgs additi
         list(APPEND HEADERS_NO_INSTALL ${GLOB_FILES})
     endforeach()
 
+endmacro()
+
+macro(_facelift_add_target_start ACTUAL_TARGET_NAME)
```
Comment (bitmouse) [inline cmake/faceliftMacros.cmake]: [minor] maybe CURRENT_TARGET_NAME would be better name here?

File: cmake/faceliftMacros.cmake [hunk 4]
```diff
@@ -403,57 +414,130 @@ macro(_facelift_add_target_start additionalOptions additionalOneValueArgs additi
...
         endif()
-        add_library(${TARGET_NAME} ${LIBRARY_TYPE} ${ALL_SOURCES})
-        if(ENABLE_LTO)
-           if(NOT "${LIBRARY_TYPE}" STREQUAL "INTERFACE")
-               set_property(TARGET ${TARGET_NAME} PROPERTY INTERPROCEDURAL_OPTIMIZATION TRUE)
-           endif()
+
+        if(MONOLITHIC_BUILD AND ${ARGUMENT_MONOLITHIC_SUPPORTED})
+
+#            message(WARNING "Enabling MONOLITHIC for lib : ${TARGET_NAME}")
```
Comment (bitmouse) [inline cmake/faceliftMacros.cmake]: Please remove commented out line or add reason why it is commented out.
Personally I would keep info message here.

File: cmake/faceliftMacros.cmake [hunk 5]
```diff
@@ -403,57 +414,130 @@ macro(_facelift_add_target_start additionalOptions additionalOneValueArgs additi
...
+
+            foreach(MONOLITHIC_LINK_LIBRARY ${ARGUMENT_MONOLITHIC_LINK_LIBRARIES})
+                list(APPEND ARGUMENT_LINK_LIBRARIES ${MONOLITHIC_LINK_LIBRARY}_)
+            endforeach()
+
+            list(APPEND ARGUMENT_LINK_LIBRARIES ${MONOLITHIC_LINK_LIBRARIES})
+            _facelift_add_target_finish(${TARGET_NAME} ${IMPLEMENTATION_TARGET_NAME})
+
+            set_property(GLOBAL APPEND PROPERTY MONOLITHIC_LIBRARIES ${IMPLEMENTATION_TARGET_NAME})
+#            get_property(source_list GLOBAL PROPERTY source_list_property)
```
Comment (bitmouse) [inline cmake/faceliftMacros.cmake]: Same as above: remove or add reason

File: cmake/faceliftMacros.cmake (hunk 4 above)
Comment (jacky309) [reply cmake/faceliftMacros.cmake]: Removed

File: cmake/faceliftMacros.cmake (hunk 5 above)
Comment (jacky309) [reply cmake/faceliftMacros.cmake]: Removed

**Pull request merged**